lazy_theta_star_p_planner/src/lazy_theta_star_p_planner.cpp

```diff
@@ -0,0 +1,190 @@
+{
+void LazyThetaStarP_Planner::configure(
+  rclcpp_lifecycle::LifecycleNode::SharedPtr parent,
+  std::string name, std::shared_ptr<tf2_ros::Buffer> tf,
+  std::shared_ptr<nav2_costmap_2d::Costmap2DROS> costmapRos)
+{
+  planner_ = std::make_unique<lazyThetaStarP::LazyThetaStarP>();
+  node_ = parent;
+  name_ = name;
+  tf_ = tf;
+  planner_->costmap_ = costmapRos->getCostmap();
+  globalFrame_ = costmapRos->getGlobalFrameID();
+
+  nav2_util::declare_parameter_if_not_declared(
+    node_, name_ + ".how_many_corners", rclcpp::ParameterValue(
+      8));
+
+  node_->get_parameter(name_ + ".how_many_corners", planner_->how_many_corners_);
+
+  if (planner_->how_many_corners_ != 8 && planner_->how_many_corners_ != 4) {
+    planner_->how_many_corners_ = 8;
+    RCLCPP_WARN(node_->get_logger(),
+      "Your value for - .how_many_coreners  was overriden, and is now set to 8");
+  }
+
+
+  nav2_util::declare_parameter_if_not_declared(
+    node_, name_ + ".interpolation_dist", rclcpp::ParameterValue(
+      0.1));
+
+  node_->get_parameter(name_ + ".interpolation_dist", interpolation_dist_);
+
+///   could be added in the future verions if required, this
+///   includes the parameters for lethal_cost, parameters for distance function and the costmap traversal costs
+//  nav2_util::declare_parameter_if_not_declared(
+//    node_, name_ + ".lethal_cost", rclcpp::ParameterValue(
+//      252));
+//
+//  node_->get_parameter(name_ + ".lethal_cost", lethal_cost_);
+//
+//    nav2_util::declare_parameter_if_not_declared(
+//    node_, name_ + ".euc_tolerance", rclcpp::ParameterValue(1.0));
+//
+//    node_->get_parameter(name_ + ".euc_tolerance", planner_->euc_tolerance_);
+//
+//    nav2_util::declare_parameter_if_not_declared(
+//    node_, name_ + ".costmap_tolerance", rclcpp::ParameterValue(1.0));
+//
+//    node_->get_parameter(name_ + ".costmap_tolerance", planner_->costmap_tolerance_);
+}
+
+void LazyThetaStarP_Planner::cleanup()
+{
+  RCLCPP_INFO(
+    node_->get_logger(), "CleaningUp plugin %s of type LazyThetaStarP_Planner",
+    name_.c_str());
+}
+
+void LazyThetaStarP_Planner::activate()
+{
+  RCLCPP_INFO(
+    node_->get_logger(), "Activating plugin %s of type LazyThetaStarP_Planner",
+    name_.c_str());
+}
+
+void LazyThetaStarP_Planner::deactivate()
+{
+  RCLCPP_INFO(
+    node_->get_logger(), "Deactivating plugin %s of type LazyThetaStarP_Planner",
+    name_.c_str());
+}
+
+nav_msgs::msg::Path LazyThetaStarP_Planner::createPlan(
+  const geometry_msgs::msg::PoseStamped & start,
+  const geometry_msgs::msg::PoseStamped & goal)
+{
+  nav_msgs::msg::Path global_path;
+  setStartAndGoal(start, goal);
+  RCLCPP_INFO(
+    node_->get_logger(), "Got the src and dst... (%i, %i) && (%i, %i)",
+    planner_->src.x, planner_->src.y, planner_->dst.x, planner_->dst.y);
+  getPlan(global_path);
+  return global_path;
+}
+
+void LazyThetaStarP_Planner::getPlan(nav_msgs::msg::Path & global_path)
+{
+  planner_->node_ = node_;
```

Comment:
Why does the planner need the node?

Reply:
That was there to print out the values, nothing else, it has to be removed